geode-core/src/main/java/org/apache/geode/redis/internal/executor/sortedset/GeoRadiusExecutor.java

```diff
@@ -0,0 +1,132 @@
+
+  @Override
+  public void executeCommand(Command command, ExecutionHandlerContext context) {
+    List<byte[]> commandElems = command.getProcessedCommand();
+
+    if (commandElems.size() < 6) {
+      command.setResponse(
+          Coder.getErrorResponse(context.getByteBufAllocator(), RedisConstants.ArityDef.GEORADIUS));
+      return;
+    }
+
+    ByteArrayWrapper key = command.getKey();
+    checkDataType(key, RedisDataType.REDIS_SORTEDSET, context);
+    Region<ByteArrayWrapper, ByteArrayWrapper> keyRegion = getRegion(context, key);
+
+    if (keyRegion == null) {
+      command.setResponse(Coder.getEmptyArrayResponse(context.getByteBufAllocator()));
+      return;
+    }
+
+    GeoRadiusParameters params;
+    try {
+      params = new GeoRadiusParameters(keyRegion, commandElems,
+          GeoRadiusParameters.CommandType.GEORADIUS);
+    } catch (IllegalArgumentException e) {
+      command.setResponse(
+          Coder.getErrorResponse(context.getByteBufAllocator(), ERROR_INVALID_ARGUMENT_UNIT_NUM));
+      return;
+    } catch (RedisCommandParserException e) {
+      command.setResponse(Coder.getErrorResponse(context.getByteBufAllocator(),
+          RedisConstants.ArityDef.GEORADIUS));
+      return;
+    } catch (CoderException e) {
+      command.setResponse(
+          Coder.getErrorResponse(context.getByteBufAllocator(), ERROR_INVALID_LATLONG));
+      return;
+    } catch (MemberNotFoundException e) {
+      /* Not possible for GEORADIUS */
+      return;
+    }
+
+    HashNeighbors hn;
+    try {
+      hn = GeoCoder.geoHashGetAreasByRadius(params.lon, params.lat, params.radius);
+    } catch (CoderException e) {
+      command.setResponse(Coder.getErrorResponse(context.getByteBufAllocator(), ERROR_NOT_NUMERIC));
+      return;
+    }
+
+    List<GeoRadiusResponseElement> results = new ArrayList<>();
+    for (String neighbor : hn.get()) {
+      try {
+        List<StructImpl> range = getGeoRadiusRange(context, key, neighbor);
+        for (StructImpl point : range) {
+          String name = point.get("key").toString();
+          char[] hashBits = point.get("value").toString().toCharArray();
+
+          Double dist = GeoCoder.geoDist(params.centerHashPrecise, hashBits) * params.distScale;
+
+          // Post-filter for accuracy
+          if (dist > (params.radius * params.distScale))
+            continue;
+
+          Optional<GeoCoord> coord =
+              params.withCoord ? Optional.of(GeoCoder.geoPos(hashBits)) : Optional.empty();
+          Optional<String> hash =
+              params.withHash ? Optional.of(GeoCoder.bitsToHash(hashBits)) : Optional.empty();
+
+          results.add(new GeoRadiusResponseElement(name, coord, dist, params.withDist, hash));
+        }
+      } catch (Exception e) {
+        command.setResponse(Coder.getErrorResponse(context.getByteBufAllocator(), e.getMessage()));
+        return;
+      }
+    }
+
+    if (params.ascendingOrder != null && params.ascendingOrder) {
```

Comment:
I would add a comment on params saying that it can be Ascending, Descending or null (unsorted) -- or maybe even use an enum for those three possibilities.

Reply:
Yep, sounds good.